src/jorphan/src/main/java/org/apache/jorphan/util/StringWrap.java

```diff
@@ -0,0 +1,166 @@
+
+    private final BreakCursor wordCursor = new BreakCursor(BreakIterator.getLineInstance());
+    private final BreakCursor charCursor = new BreakCursor(BreakIterator.getCharacterInstance());
+
+    /**
+     * Stores the current and the next position for a given {@link BreakIterator}.
+     * It allows reducing the number of calls to {@link BreakIterator}.
+     */
+    private static class BreakCursor {
+        private static final int UNINITIALIZED = -2;
+
+        private final BreakIterator iterator;
+        private int pos;
+        private int next;
+
+        BreakCursor(BreakIterator iterator) {
+            this.iterator = iterator;
+        }
+
+        void setText(String text) {
+            iterator.setText(text);
+            pos = 0;
+            next = UNINITIALIZED;
+        }
+
+        public int getPos() {
+            return pos;
+        }
+
+        /**
+         * Advances the cursor if possible.
+         * @param startWrap the start index of the wrap to consider
+         * @param endWrap the end index of the wrap to consider
+         * @return true if the next break is detected within startWrap..endWrap boundaries
+         */
+        public boolean advance(int startWrap, int endWrap) {
+            if (pos == BreakIterator.DONE || pos > endWrap) {
+                return false;
+            }
+            pos = next != UNINITIALIZED ? next : iterator.following(startWrap);
+            if (pos == BreakIterator.DONE || pos > endWrap) {
+                return false;
+            }
+            // Try adding more items up to endWrap
+            while (true) {
+                next = iterator.next();
+                if (next == BreakIterator.DONE || next > endWrap) {
+                    break;
+                }
+                pos = next;
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Creates string wrapper instance.
+     *
+     * @param minWrap minimal word length for the wrap
+     * @param maxWrap maximum word length for the wrap
+     */
+    public StringWrap(int minWrap, int maxWrap) {
+        this.minWrap = minWrap;
+        this.maxWrap = maxWrap;
+    }
+
+    public int getMinWrap() {
+        return minWrap;
+    }
+
+    public int getMaxWrap() {
+        return maxWrap;
+    }
+
+    /**
+     * Wraps given {@code input} text accoding to
+     *
+     * @param input     input text
+     * @param delimiter delimiter when inserting soft wraps
+     * @return modified text with added soft wraps, or input if wraps are not needed
+     */
+    public String wrap(String input, String delimiter) {
+        if (input.length() <= minWrap) {
+            return input;
+        }
+        wordCursor.setText(input);
+        charCursor.setText(input);
+        int pos = 0;
+        StringBuilder sb = new StringBuilder(input.length() + input.length() / minWrap * delimiter.length());
+        boolean hasChanges = false;
+        int nextLineSeparator = -2;
```

Comment:
Is this the constant `BreakCursor.UNITIALIZED`?

Reply:
I just made up this constant

Reply:
ah, indeed, this should be `BreakCursor.UNITIALIZED`